src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const keys in actions[i].extraData) {
+          state[keys] = actions[i].extraData[keys];
+        }
+        break;
+
+      case 'removeProperties':
+        for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+          delete state[actions[i].keysToRemove[y]];
+        }
```

Comment:
You can use for of here
```suggestion
        for (const key of actions[i].keysToRemove) {
          delete state[key];
        }
```